dnf-behave-tests/dnf/plugins-core/modulesync.feature

```diff
@@ -0,0 +1,95 @@
+@1868047
+Feature: Test for modulesync command
+
+
+Background:
+  Given I use repository "dnf-ci-fedora-modular"
+    And I use repository "dnf-ci-fedora"
+    And I enable plugin "modulesync"
+
+
+Scenario: I can download a module
+  Given I create directory "modulesync"
+    And I set working directory to "{context.dnf.installroot}/modulesync"
+    And I execute dnf with args "modulesync nodejs:8/minimal"
+   Then the exit code is 0
+    And file "modulesync/nodejs-8.11.4-1.module_2030+42747d40.src.rpm" exists
+    And file "modulesync/nodejs-8.11.4-1.module_2030+42747d40.x86_64.rpm" exists
+    And file "modulesync/nodejs-devel-8.11.4-1.module_2030+42747d40.x86_64.rpm" exists
+    And file "modulesync/nodejs-docs-8.11.4-1.module_2030+42747d40.noarch.rpm" exists
+    And file "modulesync/npm-8.11.4-1.module_2030+42747d40.x86_64.rpm" exists
+    And stdout is
+    """
+    <REPOSYNC>
+    Directory walk started
+    Directory walk done - 5 packages
+    Loaded information about 0 packages
+    Temporary output repo path: {context.dnf.installroot}/modulesync/.repodata/
+    Preparing sqlite DBs
+    Pool started (with 5 workers)
+    Pool finished
+    """
+
+
+Scenario: I can download a module with option newest only
+  Given I use repository "dnf-ci-fedora-modular-updates"
+    And I create directory "modulesync"
+    And I set working directory to "{context.dnf.installroot}/modulesync"
+    And I execute dnf with args "modulesync nodejs:8/minimal --newest-only"
+   Then the exit code is 0
+    And file "modulesync/nodejs-8.11.4-1.module_2030+42747d40.src.rpm" exists
+    And file "modulesync/nodejs-8.11.4-1.module_2030+42747d40.x86_64.rpm" exists
+    And file "modulesync/nodejs-devel-8.11.4-1.module_2030+42747d40.x86_64.rpm" exists
+    And file "modulesync/nodejs-docs-8.11.4-1.module_2030+42747d40.noarch.rpm" exists
+    And file "modulesync/npm-8.14.0-1.module_2030+42747d41.x86_64.rpm" exists
+    And stdout is
+    """
+    <REPOSYNC>
+    Directory walk started
+    Directory walk done - 5 packages
+    Loaded information about 0 packages
+    Temporary output repo path: {context.dnf.installroot}/modulesync/.repodata/
+    Preparing sqlite DBs
+    Pool started (with 5 workers)
+    Pool finished
+    """
+
+
+Scenario: I can download a module and init a modular repository with modulesync
+  Given I create directory "modulesync"
+    And I set working directory to "{context.dnf.installroot}/modulesync"
+    And I successfully execute dnf with args "install nodejs --downloadonly --downloaddir=."
```

Comment:
What about to use the scenario to test proper functionality of `--downloaddir`? Then I recommend to skip line 60 
```    And I set working directory to "{context.dnf.installroot}/modulesync"```.
Modify line 61 `And I successfully execute dnf with args "install nodejs --downloadonly --downloaddir={context.dnf.installroot}/modulesync"`. Then modify line 70 `When I execute dnf with args "modulesync --downloaddir={context.dnf.installroot}/modulesync"`

Reply:
I am not against it, but it would result in a big scenario testing multiple things. Would you rather split it in 2 scenarios?